pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -0,0 +1,532 @@
+            )
+
+    References
+    ----------
+    .. [1] David C. Schmittlein, Donald G. Morrison and Richard Colombo.
+           "Counting Your Customers: Who Are They and What Will They Do Next."
+           Management Science,Vol. 33, No. 1 (Jan., 1987), pp. 1-24.
+    .. [2] Fader, Peter & G. S. Hardie, Bruce (2005).
+           "A Note on Deriving the Pareto/NBD Model and Related Expressions."
+           http://brucehardie.com/notes/009/pareto_nbd_derivations_2005-11-05.pdf
+    .. [3] Fader, Peter & G. S. Hardie, Bruce (2014).
+           "Additional Results for the Pareto/NBD Model."
+           https://www.brucehardie.com/notes/015/additional_pareto_nbd_results.pdf
+    .. [4] Fader, Peter & G. S. Hardie, Bruce (2014).
+           "Deriving the Conditional PMF of the Pareto/NBD Model."
+           https://www.brucehardie.com/notes/028/pareto_nbd_conditional_pmf.pdf
+    """
+
+    _model_name = "Pareto/NBD"  # Pareto Negative-Binomial Distribution
+    _params = ["r", "alpha", "s", "beta"]
+
+    def __init__(
+        self,
+        customer_id: Union[np.ndarray, pd.Series],
+        recency: Union[np.ndarray, pd.Series, TensorVariable],
+        frequency: Union[np.ndarray, pd.Series, TensorVariable],
+        T: Union[np.ndarray, pd.Series, TensorVariable],
+        r_prior: Optional[TensorVariable] = None,
+        alpha_prior: Optional[TensorVariable] = None,
+        s_prior: Optional[TensorVariable] = None,
+        beta_prior: Optional[TensorVariable] = None,
+    ):
+        warnings.warn(
+            "The Pareto/NBD model is still experimental. Please see code examples in\
+                        documentation if model fitting issues are encountered.",
+            UserWarning,
+        )
+
+        if len(np.unique(customer_id)) != len(customer_id):
+            raise ValueError("Customers must have unique ID labels.")
+
+        super().__init__()
+
+        self._customer_id = customer_id
+        self._frequency = frequency
+        self._recency = recency
+        self._T = T
+
+        r_prior, alpha_prior, s_prior, beta_prior = self._process_priors(
+            r_prior, alpha_prior, s_prior, beta_prior
+        )
+        # TODO: rename hyperpriors to purchase_shape, purchase_scale, churn_shape, churn_scale?
+        coords = {"customer_id": self._customer_id}
+        with pm.Model(coords=coords) as self.model:
+            # purchase rate priors
+            r = self.model.register_rv(r_prior, name="r")
+            alpha = self.model.register_rv(alpha_prior, name="alpha")
+
+            # churn priors
+            s = self.model.register_rv(s_prior, name="s")
+            beta = self.model.register_rv(beta_prior, name="beta")
+
+            ParetoNBD(
+                name="likelihood",
+                r=r,
+                alpha=alpha,
+                s=s,
+                beta=beta,
+                T=self._T,
+                observed=np.stack((self._recency, self._frequency), axis=1),
```

Comment:
If you want to support TensorVariables (as suggested by the type hints), you should use `pt.stack`
```suggestion
                observed=np.stack((self._recency, self._frequency), axis=1),
```